core/src/replay_stage.rs

```diff
@@ -2406,10 +2406,28 @@ impl ReplayStage {
         }
 
         let mut update_commitment_cache_time = Measure::start("update_commitment_cache");
+        // Send (voted) bank along with the updated vote account state for this node, the vote
+        // state is always newer than the one in the bank by definition, because banks can't
+        // contain vote transactions which are voting on its own slot.
+        //
+        // It should be acceptable to aggressively use the vote for our own _local view_ of
+        // commitment aggregation, although it's not guaranteed that the new vote transaction is
+        // observed by other nodes at this point.
+        //
+        // The justification stems from the assumption of the sensible voting behavior from the
+        // consensus subsystem. That's because it means there would be a slashing possibility
+        // otherwise.
+        //
+        // This behavior isn't significant normally for mainnet-beta, because staked nodes aren't
```

Comment:
i'm fine with this change. as you pointed out this has the possibility of causing mismatches when running staked rpc nodes, however unless our node has > 1/6 of the stake the block will become a supermajority root anyway. And in the case where it fails the cluster will halt unless our node has > 1/3 of the stake, so a non issue.